File: android/src/main/java/com/github/quadflask/react/navermap/RNNaverMapFeature.java
package com.github.quadflask.react.navermap;

import android.content.Context;

import com.naver.maps.map.overlay.Overlay;

public abstract class RNNaverMapFeature<T extends Overlay> extends EventEmmitableReactView {
    protected T feature;

    public RNNaverMapFeature(EventEmittable emitter, Context context) {
        super(emitter, context);
    }

    public void addToMap(RNNaverMapView map) {
        feature.setMap(map.getMap());
    }

    public void removeFromMap() {
        feature.setMap(null);
        feature = null;
    }

    public T getFeature() {
        return feature;
    }
}

File: android/src/main/java/com/github/quadflask/react/navermap/feature/RNNaverMapFeature.kt
package com.github.quadflask.react.navermap.feature

import android.content.Context
import android.util.Log
import com.naver.maps.map.NaverMap
import com.naver.maps.map.overlay.Overlay

abstract class RNNaverMapFeature<T : Overlay>(
  emitter: EventEmittable,
  context: Context
) : EventEmittableReactView(emitter, context) {
  lateinit var feature: T
    protected set

  open fun addToMap(map: NaverMap) {
    feature.map = map
    Log.e("RNN", "RNNaverMapFeature.addToMap")
  }

  fun removeFromMap() {
    feature.map = null
    Log.e("RNN", "RNNaverMapFeature.removeFromMap")
  }
}
